api/src/repositories/history-publish-repository.ts

```diff
@@ -609,10 +613,221 @@ export class HistoryPublishRepository extends BaseRepository {
    */
   async deleteSurveyReportAttachmentPublishRecord(surveyAttachmentId: number): Promise<void> {
     const sqlStatement = SQL`
-      delete from survey_report_publish
-      where survey_report_attachment_id = ${surveyAttachmentId};
+      delete
+      from
+        survey_report_publish
+      where
+        survey_report_attachment_id = ${surveyAttachmentId};
     `;
 
     await this.connection.sql(sqlStatement);
   }
+
+  /**
+   * Gets the count of unpublished survey attachments
+   *
+   * @param {number} surveyId
+   * @return {*}  {Promise<QueryResult>}
+   * @memberof HistoryPublishRepository
+   */
+  async getCountSurveyUnpublishedAttachments(surveyId: number): Promise<QueryResult> {
+    const sqlStatement = SQL`
+    SELECT
+      count(*)
+    from
+      survey_attachment sa
+    left join
+      survey_attachment_publish sap
+    on
+      sa.survey_attachment_id = sap.survey_attachment_id
+    where
+      sa.survey_id =${surveyId}
+    and
+      sap.survey_attachment_publish_id is null;
+  `;
+
+    const response = await this.connection.sql(sqlStatement);
+
+    return response;
+  }
+
+  /**
+   * Gets the count of unpublished survey reports
+   *
+   * @param {number} surveyId
+   * @return {*}  {Promise<QueryResult>}
+   * @memberof HistoryPublishRepository
+   */
+  async getCountSurveyUnpublishedReports(surveyId: number): Promise<QueryResult> {
+    const sqlStatement = SQL`
+    SELECT
+      count(*)
+    from
+      survey_report_attachment sra
+    left join
+      survey_report_publish srp
+    on
+      sra.survey_report_attachment_id = srp.survey_report_attachment_id
+    where
+      sra.survey_id =${surveyId}
+    and
+      srp.survey_report_publish_id is null;
+  `;
+
+    const response = await this.connection.sql(sqlStatement);
+
+    return response;
+  }
+
+  /**
+   * Gets the latest survey observation that wasn't deleted
+   *
+   * @param {number} surveyId
+   * @return {*}  {Promise<QueryResult>}
+   * @memberof HistoryPublishRepository
+   */
+  async getLatestUndeletedObservationRecordId(surveyId: number): Promise<QueryResult> {
+    const sqlStatement = SQL`
+    select
+      occurrence_submission_id
+    from
+      occurrence_submission os
+    where
+      os.survey_id = ${surveyId}
+    and
+      os.delete_timestamp is null
+    order by
+      event_timestamp desc
+    limit 1;`;
+    const response = await this.connection.sql(sqlStatement);
+
+    console.log('response: ', response);
+
+    return response;
+  }
+
+  /**
+   * Gets a publication record for an observation occurrence submission
+   *
+   * @param {number} occurrenceSubmissionId
+   * @return {*}  {Promise<QueryResult>}
+   * @memberof HistoryPublishRepository
+   */
+  async getConfirmationLatestObservationPublished(occurrenceSubmissionId: number): Promise<QueryResult> {
```

Comment:
I think this is a duplicate of `getOccurrenceSubmissionPublishRecord`

Reply:
swapped out